core/sds-aspect-model-java-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/java/JavaCodeGenerationConfig.java

```diff
@@ -22,22 +23,70 @@
    private final boolean enableJacksonAnnotations;
    private final String packageName;
    private final ImportTracker importTracker;
+   private final boolean executeLibraryMacros;
+   private final String templateLibPath;
+   private final String templateLibFileName;
 
-   public JavaCodeGenerationConfig( final boolean enableJacksonAnnotations, final String packageName ) {
+   public JavaCodeGenerationConfig( final boolean enableJacksonAnnotations, final String packageName, final boolean executeLibraryMacros,
+         final String templateLibPath, final String templateLibFileName ) {
```

Comment:
I'd prefer to have the path and filename passed as either one `java.io.File` or `java.nio.Path` instead of Strings.